docs/rules/prefer-arrow-callback.md

```diff
@@ -37,6 +37,27 @@ foo(function() { return this.a; });
 foo(function bar(n) { return n && n + bar(n - 1); });
 ```
 
+## Options
+
+`prefer-arrow-callback` receives a single, optional argument which is an options object.
+
+### functionsWithThis
+
+This defaults to `true`. When set to `false` it disallows using "dynamic" `this` in callback functions where `this` is not bound using `bind` and determined by the higher-level function.
```

Comment:
I don't think "dynamic" is useful here, especially since it is not further explained. I'd say something like:

> when set to false, callbacks that use `this` without also using `bind` are allowed. In that case, the value of `this` is not known until execution time so changing to an arrow function must be done carefully to avoid errors.